test-case/check-alsa-conformance.sh

```diff
@@ -0,0 +1,255 @@
+  test_suites=$([ "${OPT_VAL['T']}" ] && echo "--test-suites ${OPT_VAL['T']}")
+
+  AUDIOTEST_OUT="${LOG_ROOT}/alsa_conformance"
+  RESULT_JSON="${LOG_ROOT}/test_result.json"
+
+  ALSA_CONFORMANCE_OPTIONS=("${formats}" "${rates}" "${rate_diff}" "${rate_err}" "${avail_delay}" "${test_suites}")
+  PLAYBACK_DEVICE="${pcm_p}"
+  CAPTURE_DEVICE="${pcm_c}"
+  PLAYBACK_OPTIONS=""
+  CAPTURE_OPTIONS=""
+
+  ALSA_CONFORMANCE_PATH=$([ "$ALSA_CONFORMANCE_PATH" ] || realpath "${TESTDIR}/../audiotest")
+  ALSA_CONFORMANCE_TEST="${ALSA_CONFORMANCE_PATH}/alsa_conformance_test"
+}
+
+check_alsa_conformance_suite()
+{
+  if [ -d "${ALSA_CONFORMANCE_PATH}" ]; then
+    if [ -x "${ALSA_CONFORMANCE_TEST}" ] && [ -x "${ALSA_CONFORMANCE_TEST}.py" ]; then
+      dlogi "Use ALSA conformance test suite: ${ALSA_CONFORMANCE_TEST}"
+      return
+    fi
+  fi
+  skip_test "ALSA conformance test suite is missing at: ${ALSA_CONFORMANCE_PATH}"
+}
+
+check_options()
+{
+  if [ "${alsa_device}" ]; then
+    if [ "${pcm_p}" ] || [ "${pcm_c}" ]; then
+      die "Give either ALSA device, or ALSA playback/capture PCMs."
+    fi
+    PLAYBACK_DEVICE="${alsa_device}"
+    PLAYBACK_OPTIONS="-P ${alsa_device}"
+    CAPTURE_DEVICE="${alsa_device}"
+    CAPTURE_OPTIONS="-C ${alsa_device}"
+  elif [ -z "${pcm_p}" ] && [ -z "${pcm_c}" ]; then
+      die "Neither playback, nor capture ALSA PCM is specified."
+  else
+    [ "${pcm_p}" ] && PLAYBACK_OPTIONS="-P ${pcm_p}"
+    [ "${pcm_c}" ] && CAPTURE_OPTIONS="-C ${pcm_c}"
+  fi
+}
+
+set_alsa()
+{
+  reset_sof_volume
+
+  # If MODEL is defined, set proper gain for the platform
+  if [ -z "$MODEL" ]; then
+    dlogw "No MODEL is defined. Please define MODEL to run alsa_settings/\${MODEL}.sh"
+  else
+    set_alsa_settings "$MODEL"
+  fi
+}
+
+alsa_conformance_device_info()
+{
+  local run_cmd="${ALSA_CONFORMANCE_TEST} ${PLAYBACK_OPTIONS} ${CAPTURE_OPTIONS} --dev_info_only"
+  dlogc "${run_cmd}"
+  local rc=1
+  bash -c "${run_cmd}" ; rc=$?
+  [[ "${rc}" -ne 0 ]] && die "Failed to get device info, rc=${rc}"
+}
+
+alsa_conformance_test_playback()
+{
+  if [ "${PLAYBACK_OPTIONS}" ]; then
+    local run_cmd="PATH=${ALSA_CONFORMANCE_PATH}:${PATH} "
+    run_cmd+="${ALSA_CONFORMANCE_TEST}.py ${ALSA_CONFORMANCE_OPTIONS[*]} ${PLAYBACK_OPTIONS} "
+    run_cmd+="--json-file ${AUDIOTEST_OUT}_playback.json ${run_verbose}"
+    dlogc "${run_cmd}"
+    local rc=1
+    bash -c "${run_cmd}" ; rc=$?
+    [[ "${rc}" -ne 0 ]] && dloge "Failed PLAYBACK TESTS, rc=${rc}"
+  fi
+}
+
+alsa_conformance_test_capture()
+{
+  if [ "${CAPTURE_OPTIONS}" ]; then
+    local run_cmd="PATH=${ALSA_CONFORMANCE_PATH}:${PATH} "
+    run_cmd+="${ALSA_CONFORMANCE_TEST}.py ${ALSA_CONFORMANCE_OPTIONS[*]} ${CAPTURE_OPTIONS} "
+    run_cmd+="--json-file ${AUDIOTEST_OUT}_capture.json ${run_verbose}"
+    dlogc "${run_cmd}"
+    local rc=1
+    bash -c "${run_cmd}" ; rc=$?
+    [[ "${rc}" -ne 0 ]] && dloge "Failed CAPTURE TESTS, rc=${rc}"
```

Comment:
```suggestion
    bash -c "${run_cmd}" || dloge "Failed CAPTURE TESTS, rc=$?"
```
BTW shouldn't this fail the test? `dloge` is not enough AFAIK.